### PR TITLE
Added functional tests

### PR DESCRIPTION
**Pull request by leobcgov**
# Fixes or Implements [AB#9176](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9176)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Description
- Verified existing and Added functional tests

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

NO

### Browsers Tested

* [ ] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.  

</br>
</br>
</br>
___
  
## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"?

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

### CODE REVIEW

File: Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
 const { AuthMethod } = require("../../support/constants")
 
 describe('Health Insights', () => {
-    beforeEach(() => {
+    before(() => {
         cy.login(Cypress.env('keycloak.username'), 
                  Cypress.env('keycloak.password'), 
                  AuthMethod.KeyCloak)
+        cy.get('#covid-modal___BV_modal_header_ > .close').click()
```
Comment (sslaws) [inline Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js:8]: Doesn’t the modal have a dataid?  Also, why not stub the config to turn off the modal?  We should also think about making a story to preserve the route path on login.

File: Apps/WebClient/test/functional/e2e/cypress/integration/user/profile.js [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
                  Cypress.env('keycloak.password'), 
                  AuthMethod.KeyCloak)
+        cy.get('#covid-modal___BV_modal_header_ > .close').click()
+        cy.get('#menuBtnProfile').click()
     })
 
-    it('Test1', () => {
-
+    it('Profile - Email Validation - email fields should be identical', () => {
+        cy.get('#editEmail').click()
```
Comment (sslaws) [inline Apps/WebClient/test/functional/e2e/cypress/integration/user/profile.js:14]: No test Id?

**Review by sslaws — CHANGES_REQUESTED** [review]
We should be using test ids globally, I commented on a few.

File: Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js (hunk 1 above)
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js:8]: doesn't seem we can add dataid to the close button, but I will check it out. stub the config is interesting idea. Can you expand a bit on how to do so?

File: Apps/WebClient/test/functional/e2e/cypress/integration/user/profile.js (hunk 2 above)
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/user/profile.js]: nice catch, I have missed this one.

**Review by tiagograf-AOT — APPROVED** [review]

**Review by FuriousLlama — APPROVED** [review]

File: Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js (hunk 1 above)
Comment (sslaws) [reply Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js]: I believe you're right but you should be able to get the modal and then fund the button within it.

**Review by FuriousLlama — APPROVED** [review]

File: Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js (hunk 1 above)
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js]: sounds good, I will use this for other tests. for this modal, we leverage the configuration to disable them wherever possible.

File: Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
                  Cypress.env('keycloak.password'), 
                  AuthMethod.KeyCloak)
+        cy.get('#menuBtnHealthInsights').click()
     })
 
-    it('Test 1', () => {
-    })
-
-    it('Test 2', () => {
+    it('totally has 259 records', () => {
```
Comment (sslaws) [inline Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js:14]: The it should document the test, that is the what.  This doesn't tell us anything.

This applies to many of your tests.

**Review by sslaws — CHANGES_REQUESTED** [review]

File: Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js (hunk 3 above)
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js]: @sslaws I have updated the test to "Total records count (for medication) should not have 0 records.", can you please review it again?

**Review by FuriousLlama — DISMISSED** [review]

File: Apps/WebClient/test/functional/e2e/cypress/integration/pages/login/KeycloakLogin.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const { AuthMethod } = require("../../../support/constants")
+
+describe('Keycloak Login', () => {
+    before(() => {
+        cy.login(Cypress.env('keycloak.username'), 
+                 Cypress.env('keycloak.password'), 
+                 AuthMethod.KeyCloak)
+        cy.get('#covid-modal___BV_modal_header_ > .close').click()
```
Comment (tiagograf-AOT) [inline Apps/WebClient/test/functional/e2e/cypress/integration/pages/login/KeycloakLogin.js:8]: this will fail if the covid modal doesn't display

Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/pages/login/KeycloakLogin.js]: removed covid modal

**Review by tiagograf-AOT — DISMISSED** [review]

File: Apps/WebClient/test/functional/e2e/cypress/integration/pages/login/KeycloakLoginProtected.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+const { AuthMethod } = require("../../../support/constants")
+
```
Comment (sslaws) [inline Apps/WebClient/test/functional/e2e/cypress/integration/pages/login/KeycloakLoginProtected.js:2]: This test was already written as well.

File: Apps/WebClient/test/functional/e2e/cypress/integration/user/registration.js [hunk 6]
```diff
@@ -2,15 +2,11 @@ const { AuthMethod } = require("../../support/constants")
...
+        cy.login(Cypress.env('keycloak.accountclosure.username'), 
                  Cypress.env('keycloak.password'), 
                  AuthMethod.KeyCloak)
     })
 
-    it('Test1', () => {
-
-    })
-
     it('Test2', () => {
```
Comment (sslaws) [inline Apps/WebClient/test/functional/e2e/cypress/integration/user/registration.js:10]: Are we testing something here?

**Review by sslaws — CHANGES_REQUESTED** [review]

File: Apps/WebClient/test/functional/e2e/cypress/fixtures/insightsConfig.json [hunk 7]
```diff
@@ -0,0 +1,67 @@
+{
```
[the remark this answers is not in the document]
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/fixtures/insightsConfig.json:1]: per our call, I will spend sometime on this to find a way to change config in the code.

File: Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js [hunk 8]
```diff
@@ -1,15 +1,17 @@
 const { AuthMethod } = require("../../support/constants")
 
 describe('Health Insights', () => {
-    beforeEach(() => {
+    before(() => {
```
[the remark this answers is not in the document]
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/healthinsights/insights.js:4]: before is to run code once then chain the tests.

File: Apps/WebClient/test/functional/e2e/cypress/integration/pages/login/KeycloakLogin.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+const { AuthMethod } = require("../../../support/constants")
```
[the remark this answers is not in the document]
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/pages/login/KeycloakLogin.js:1]: removed

File: Apps/WebClient/test/functional/e2e/cypress/integration/pages/login/KeycloakLoginProtected.js (hunk 5 above)
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/pages/login/KeycloakLoginProtected.js:2]: removed

File: Apps/WebClient/test/functional/e2e/cypress/integration/user/registration.js (hunk 6 above)
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/integration/user/registration.js:14]: confirmed, nothing for this test

File: Apps/WebClient/test/functional/e2e/cypress/fixtures/insightsConfig.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (tiagograf-AOT) [reply Apps/WebClient/test/functional/e2e/cypress/fixtures/insightsConfig.json:1]: you can refer to my PR, I have an example there

**Review by FuriousLlama — DISMISSED** [review]

File: Apps/WebClient/test/functional/e2e/cypress/fixtures/insightsConfig.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (leobcgov) [reply Apps/WebClient/test/functional/e2e/cypress/fixtures/insightsConfig.json]: I like it, thanks @tiagograf-AOT

File: Apps/WebClient/test/functional/e2e/cypress/integration/user/profile.js [hunk 10]
```diff
@@ -1,17 +1,32 @@
...
+        cy.fixture('AllDisabledConfig').as('config')
+        cy.route('GET', '/v1/api/configuration/', '@config')
         cy.login(Cypress.env('keycloak.username'), 
                  Cypress.env('keycloak.password'), 
                  AuthMethod.KeyCloak)
     })
 
-    it('Test1', () => {
-
+    it('Profile - Email Validation - email fields should be identical', () => {
```
Comment (sslaws) [inline Apps/WebClient/test/functional/e2e/cypress/integration/user/profile.js:14]: The test name is too long, you're in Profile so we don't need to replicate that and what we're doing is Email Validation.

File: Apps/WebClient/test/functional/e2e/cypress/integration/user/profileaccountclosure.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+const { AuthMethod } = require("../../support/constants")
+
+describe('User Profile - Account Closure', () => {
```
Comment (sslaws) [inline Apps/WebClient/test/functional/e2e/cypress/integration/user/profileaccountclosure.js:3]: As above, naming

File: Apps/WebClient/test/functional/e2e/cypress/integration/user/profileaccountclosure.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+    before(() => {
+        cy.login(Cypress.env('keycloak.accountclosure.username'), 
+                 Cypress.env('keycloak.password'), 
+                 AuthMethod.KeyCloak)
+    })
+
+    it('Profile page url should contain registration', () => {
```
Comment (sslaws) [inline Apps/WebClient/test/functional/e2e/cypress/integration/user/profileaccountclosure.js:10]: again

**Review by sslaws — CHANGES_REQUESTED** [review]

**Review by sslaws — APPROVED** [review]

**Review by tiagograf-AOT — APPROVED** [review]

**Review by FuriousLlama — APPROVED** [review]

**Pull request merged**